### PR TITLE
Add API definition for ECDS listerner filter support

### PR DESCRIPTION
**Pull request by yanjunxiang-google**
Problem description:
Add API definition for ECDS listerner filter support. This is to add API change for issue: https://github.com/envoyproxy/envoy/issues/20049.

Build:
passed

Testing:
passed

Release Notes:
N/A

Issues: ecds: add support for listener filters #20049

Signed-off-by: Yanjun Xiang <yanjunxiang@google.com>

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Review by adisuissa — APPROVED** [review]
Changes LGTM, modulo pending CI checks.
I'll let @lizan take a non-googler look at this and to approve API changes.

**Review by lizan — APPROVED** [review]

**Pull request merged**
